### PR TITLE
Check requires arc true false strings

### PR DESCRIPTION
**Pull request by VinayGuthal**
This fixes https://github.com/CocoaPods/CocoaPods/issues/6701.
Add a warning if the strings` 'true'` or `'false' ` are being used instead of boolean `true` or `false`  as the value for 'requires_arc' in the pod spec file.

### CODE REVIEW

**Comment by dnkoutso** [comment]
@VinayGuthal please add `CHANGELOG` entry, run `rubocop -a` and squash to a single commit.

LGTM.

File: lib/cocoapods-core/specification/linter.rb [hunk 1]
```diff
@@ -99,6 +100,22 @@ def validate_root_name
         end
       end
 
+      # Generates a warning if the requires_arc attribute has true or false string values.
+      #
+      # @return [void]
+      #
+      def check_requires_arc_attribute
+        attributes = DSL.attributes.values
+        attributes.each do |attr|
```
Comment (dnkoutso) [inline lib/cocoapods-core/specification/linter.rb]: can you just use `attributes.find { |attr| attr.name == :requires_arc }` instead?

File: lib/cocoapods-core/specification/linter.rb [hunk 2]
```diff
@@ -99,6 +100,20 @@ def validate_root_name
         end
       end
 
+      # Generates a warning if the requires_arc attribute has true or false string values.
+      #
+      # @return [void]
+      #
+      def check_requires_arc_attribute
+        attribute = DSL.attributes.values.find{|attr| attr.name == :requires_arc}
```
Comment (dnkoutso) [inline lib/cocoapods-core/specification/linter.rb]: is the formatting correct here? I think spaces are needed, `rubocop -a` I believe will take care of it for you.

**Comment by dnkoutso** [comment]
👌 !

@DanToml or @orta quick check?

**Comment by orta** [comment]
Yep, good idea

**Comment by VinayGuthal** [comment]
Is there anything else that I need to take care of here? :)

**Comment by orta** [comment]
No, I don't think so, let's get this in.

**Pull request merged**